core/tasks.py

```diff
@@ -47,72 +52,47 @@ def validate_data(object_id, model=None):
     logger.debug("Start validation for %s file" % object_id)
 
     is_valid = False
-    try:
-        with open(datasource.file.path, "rb") as f:
-            items = ijson.items(f, "records.item")
-            for item in items:
-                if item:
-                    is_valid = True
-                    break
-        if not is_valid:
+    with open(f"{DATA_DIR}/schema.json") as schema_rd:
+        spec = DataPreprocessor(json.loads(schema_rd.read()), ROOT_TABLES, combined_tables=COMBINED_TABLES)
+        try:
+            resource = "records"
```

Comment:
we have to detect dataset type, best way to do it is to use ocdskit as library. It provides this features